### DISCUSSION TITLE
`plugin-framework` - Service methods options not showing up in the `CodeGeneratorRequest`

### DISCUSSION THREAD

**Discussion by stefanosusini** [timostamm/protobuf-ts#192] [Q&A]
Hello everyone! Thanks for your time and the great tool! 👍 

I'm trying to use the `@protobuf-ts/plugin-framework` to create a specific library for which I need to generate my own ts code.
The tool works great in providing all the files to generate and all the dependency files, so I can have the full picture when I generate the code.
My problem rises when I found to a service like the following:
```
service Service {
  rpc Get(GetRequest) returns (GetResponse) {
    option (google.api.http) = {
      get: "/v1/some-path"
    };
  }
}
```
The request contains the following data:
```
"service": [
    {
        "method": [
            {
                "name": "Get",
                "inputType": ".service.api.v1.GetRequest",
                "outputType": ".service.api.v1.GetResponse",
                "options": {
                    "uninterpretedOption": []
                }
            },
        ],
        "name": "Service"
    }
],
```
**My issue is that the request data has no mention of the `google.api.http` option and its value.**
Am I doing something wrong? (most likely 😝)
Can anyone give me some insight on this?
Thanks again!

**Comment by timostamm** [comment]
The data of the option `google.api.http` is in the method descriptor message, but not in deserialized form, but as "unknown" fields. 

Have a look at the plugin's [interpreter](https://github.com/timostamm/protobuf-ts/blob/fbce12b7259cacea5d45ebf3503c8ac8a0200c73/packages/plugin/src/interpreter.ts#L60) for an example how to read the unknown fields. What the interpreter does is actually a bit more involved than what you need. What you want is field no 72295728 (see google/api/annotations.proto), and should be able to decode the binary field data using a `protobuf-ts` generated message type for `google.api.HttpRule`.

**Reply by stefanosusini** [reply]
Thanks for your help @timostamm I'll have a look at the interpreter.
I don't understand how I get the information though, given that the service descriptor does not contain any reference to it.
If the value is not deserialized, do I need to read the proto file once again?
